### PR TITLE
Fixes due to changes in Sir structure.

### PR DESCRIPTION
**Pull request by vext01**
The Sir was changed to make way for trace interpretation:
https://github.com/softdevteam/yk/pull/30

### CODE REVIEW

File: src/librustc_yk_sections/emit_sir.rs [hunk 1]
```diff
@@ -349,10 +358,19 @@ impl<'a, 'tcx> ConvCx<'a, 'tcx> {
...
+            ty::Bool => Ok(ykpack::Constant::Int(self.lower_bool(sclr))),
             _ => Err(()), // FIXME Not implemented.
         }
     }
 
+    fn lower_bool(&mut self, sclr: &Scalar) -> ykpack::ConstantInt {
+        match sclr {
+            Scalar::Raw{data: 0, size: 1} => ykpack::ConstantInt::from(false),
+            Scalar::Raw{data: 1, size: 1} => ykpack::ConstantInt::from(true),
+            _ => panic!("bogus boolean in MIR"),
```
Comment (ptersilie) [inline src/librustc_yk_sections/emit_sir.rs]: Should we improve on that error message? ;-)

Comment (vext01) [reply src/librustc_yk_sections/emit_sir.rs]: Yeah, we certainly can!

**Comment by vext01** [comment]
Try that last commit.

**Comment by ptersilie** [comment]
I think I took more issue with the word "bogus", but at the end of the day I don't care too much, so this will do.

**Comment by vext01** [comment]
So shall I squash and add the cycle breaker commit?

**Comment by ptersilie** [comment]
Let's squash first. And then add the cycle breaker commit.

**Comment by vext01** [comment]
splat.

Ready to add cycle breaker.

**Comment by ptersilie** [comment]
Go go go!

**Comment by vext01** [comment]
Ready.

**Pull request merged**